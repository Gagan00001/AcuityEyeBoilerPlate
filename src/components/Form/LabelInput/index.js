import React from 'react';
import PropTypes from 'prop-types';
import {
  Form, Input, Row, Col,
} from 'antd';
import './index.scss';
import classNames from 'classnames';

function LabelInput({
  label,
  value,
  name,
  required,
  labelSpan,
  inputSpan,
  disabled,
  id,
  ...otherProps
}) {
  return (
    <div className="custom-input-disabled">
      <Row>
        <Col span={labelSpan}>
          <div className="ant-form-item-label ant-form-item-label-left">
            <label
              className={classNames('ant-form-item-no-colon')}
              title={label}
              htmlFor={id}
            >
              {label}
            </label>
            <span className={classNames('req-star', required && 'ant-form-item-required')} />
          </div>
        </Col>
        <Col span={inputSpan}>
          <Form.Item
            name={name}
            {...otherProps}
          >
            <Input value={value} disabled={disabled} />
          </Form.Item>
        </Col>
      </Row>
    </div>
  );
}
LabelInput.defaultProps = {
  name: '',
  label: '',
  required: false,
  disabled: true,
  labelSpan: 10,
  inputSpan: 14,
  id: 'label-input',
  value: '',
};

LabelInput.propTypes = {
  labelSpan: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  inputSpan: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  name: PropTypes.string,
  label: PropTypes.string,
  required: PropTypes.bool,
  disabled: PropTypes.bool,
  id: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]),
  value: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string,
  ]),
};

export default React.memo(LabelInput);
