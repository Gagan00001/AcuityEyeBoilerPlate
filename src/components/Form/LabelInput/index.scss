@import "../../../css/vars";
@import "../../../css/root";
@import "../../../css/mixin";

.custom-input-disabled {
  .ant-row {
    .ant-form-item-label > label {
      font-size: $font-size-xsm;
      color: $color-blue-light;
      white-space: nowrap;
      width: 11.6rem;
      overflow: hidden;
      text-overflow: ellipsis;
      display: block;
      line-height: 3.0rem;
    }
    .ant-form-item-control {
      .ant-input {
        height: 2.8rem;
        background-color: rgba(0, 0, 0, 0);
        color: $color-blue-light;
        border: 0px;
        font-weight: 500;
        padding: 0;
      }
      .ant-input:focus {
        outline: 0;
      }
    }
  }
}

.ant-col.ant-col-20.ant-form-item-control {
  max-width: 100%;
}


@media (min-width: $medium-desktop) {
  .custom-input-disabled .ant-row .ant-form-item-label > label {
    white-space: nowrap;
    width: 14.0rem;
    overflow: hidden;
    text-overflow: ellipsis;
    display: block;
  }
}
@media (min-width: $lg-desktop) {
  .custom-input-disabled .ant-row .ant-form-item-label > label {
    white-space: nowrap;
    width: 16.2rem;
    overflow: hidden;
    text-overflow: ellipsis;
    display: block;
    padding-top: 0.7rem;
  }
}
