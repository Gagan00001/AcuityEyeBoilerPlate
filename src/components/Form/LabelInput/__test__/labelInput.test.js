/* eslint-disable no-undef */
import React from 'react';
import { render } from '@testing-library/react';
import { unmountComponentAtNode } from 'react-dom';
import LabelInput from '../index';

let container = null;

beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

const labelProps = {
  required: true,
};
const RenderLabelInput = (props) => (<LabelInput {...labelProps} {...props} />);

describe('should match error snapshot', () => {
  test('should match snapshot', () => {
    const { container: labelContainer } = render(
      <RenderLabelInput />, container,
    );
    expect(labelContainer).toMatchSnapshot();
  });
});
