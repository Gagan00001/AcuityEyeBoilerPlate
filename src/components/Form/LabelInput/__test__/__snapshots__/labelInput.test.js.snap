// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match error snapshot should match snapshot 1`] = `
<div>
  <div
    class="custom-input-disabled"
  >
    <div
      class="ant-row"
    >
      <div
        class="ant-col ant-col-10"
      >
        <div
          class="ant-form-item-label ant-form-item-label-left"
        >
          <label
            class="ant-form-item-no-colon"
            for="label-input"
            title=""
          />
          <span
            class="req-star ant-form-item-required"
          />
        </div>
      </div>
      <div
        class="ant-col ant-col-14"
      >
        <div
          class="ant-row ant-form-item"
        >
          <div
            class="ant-col ant-form-item-control"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  class="ant-input ant-input-disabled"
                  disabled=""
                  id=""
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
